lib/rules/component-name-in-template-casing.js

```diff
@@ -106,15 +113,8 @@ module.exports = {
         }
         return true
       }
-      // We only verify the components registered in the component.
-      if (
-        [...registeredComponents]
-          .filter((name) => casing.isPascalCase(name)) // When defining a component with PascalCase, you can use either case
-          .some(
-            (name) =>
-              node.rawName === name || casing.pascalCase(node.rawName) === name
-          )
-      ) {
+      // We only verify the registered components.
+      if (registeredComponents.has(casing.pascalCase(node.rawName))) {
```

Comment:
I've [tested](https://stackblitz.com/edit/vitejs-vite-x9voie?file=src%2FApp.vue,src%2Fmain.js) and found that it won't work. But the original logic is also wrong because [it filters out the non-pascal-case names](https://github.com/vuejs/eslint-plugin-vue/pull/1989/files/5eb9c6945173f9a4d065e27a86c7d002cc61386e#diff-ae2b61140fbfc71296afe34f070548020a61fdf6a98e2320e47143c776da7589L112). So the bahaviour is essentially the same here. 
I think the situation is a little complicated here: if a component is registered with kebab-case, but used as PascalCase, how should our rule behave here if case style is set to PascalCase? Seems like it shouldn't report it as an error but if it doesn't, the component in fact won't be resolved and rendered correctly.(But maybe it is out of the rule's scope?)

Reply:
@ota-meshi is this logic change fine for you?

Reply:
@FloEdelmann Yeah. I think it's fine.